src/usb_stream.cpp

```diff
@@ -349,10 +349,52 @@ bool UsbStream::close_device()
 }
 
 bool UsbStream::read_device(std::vector<uint8_t>& msg, const bool stream_on)
+{
+    // read device and then store number of bytes read
+    return_value_ = UsbStream::read_maybe_device(msg, stream_on);
+    /**
+     * Check the potential error:
+     *
+     * - First we check if the port could be read at all.
+     * - Then we check if the port was read before the timeout
+     * - Then we check the validity of the message FIXME where is validity checking?
+     */
+
+    // Port reading failure
+    if (return_value_ < 0)
+    {
+        int errsv = errno;
+        rt_printf(
+            "UsbStream::read_device: "
+            "Failed to read port %s with error\n\t%s\n",
+            file_name_.c_str(),
+            strerror(errsv));
+        return false;
+    }
+    // Timeout failure
+    else if (return_value_ != static_cast<ssize_t>(msg.size()))
+    {
+        rt_printf(
+            "UsbStream::read_device: "
+            "Failed to read port %s. Requested %ld bytes and "
+            "received %ld bytes: %s\n",
+            file_name_.c_str(),
+            msg.size(),
+            return_value_,
+            msg_debug_string(buffer_, return_value_).c_str());
+        return false;
+    }
+    // Here we copy the message inside the buffer in order to use a bigger
+    // memory buffer than the message itself
+    std::copy_n(buffer_.begin(), msg.size(), msg.begin());
+    return true;
+}
+
+ssize_t UsbStream::read_maybe_device(std::vector<uint8_t>& msg, const bool stream_on, const size_t start_location)
```

Comment:
I do not get the maybe in the name here

Reply:
That was my fault. I thought it would make sense but you are right, it's not great. How about `read_device_raw`?

Reply:
sounds good.